docs/resources/webhook.md

```diff
@@ -0,0 +1,97 @@
+  ]
+}
+```
+
+<!-- schema generated by tfplugindocs -->
+## Schema
+
+### Required
+
+- `config` (Attributes) The webhook configuration used to deliver event payloads. (see [below for nested schema](#nestedatt--config))
+- `name` (String) The webhook's name.
+
+### Optional
+
+- `description` (String) The webhook's description. Descriptions are useful for helping others understand the purpose of the webhook.
+- `enabled` (Boolean) Indicates if the webhook should receive payloads for the subscribed events.
+- `project_id` (String) The ID of the project to create the webhook under. If unspecified, the webhook will be created in the project the provider is configured with.
+- `subscriptions` (Attributes List) Set of events to subscribe the webhook to all resources or a specific resource in the project. (see [below for nested schema](#nestedatt--subscriptions))
+
+### Read-Only
+
+- `resource_id` (String) The webhook's unique identifier.
+- `resource_name` (String) The webhooks's resource name in the format `webhook/project/<project_id>/geo/us/webhook/<name>`.
+
+<a id="nestedatt--config"></a>
+### Nested Schema for `config`
+
+Required:
+
+- `url` (String) The HTTP or HTTPS destination URL that HCP delivers the event payloads to. 
+The destination must be able to use the HCP webhook 
+[payload](https://developer.hashicorp.com/hcp/docs/hcp/admin/projects/webhooks#webhook-payload).
+
+Optional:
+
+- `hmac_key` (String, Sensitive) The arbitrary secret that HCP uses to sign all its webhook requests. This is awrite-only field, it is written once and not visible thereafter.
+
+
+<a id="nestedatt--subscriptions"></a>
+### Nested Schema for `subscriptions`
+
+Required:
+
+- `events` (Attributes List) The information about the events of a webhook subscription. The service that owns the resource is responsible for maintaining events. Refer to the service's webhook documentation for more information. (see [below for nested schema](#nestedatt--subscriptions--events))
+
+Optional:
+
+- `resource_id` (String) Refers to the resource the webhook is subscribed to. If not set, the webhook subscribes to the emitted events listed in events for any resource in the webhook's project.
+
+<a id="nestedatt--subscriptions--events"></a>
+### Nested Schema for `subscriptions.events`
+
+Required:
+
+- `actions` (List of String) The list of event actions subscribed for the resource type set as the [source](#source). For example, `["create", "update"]`. When the action is '*', it means that the webhook is subscribed to all event actions for the event source.
+- `source` (String) The resource type of the source of the event. For example, `hashicorp.packer.version`. Event source might not be the same type as the resource that the webhook is subscribed to ([resource_id](#resource_id)) if the event is from a descendant resource. For example, webhooks are subscribed to a `hashicorp.packer.registry` and receive events for descendent resources such as a `hashicorp.packer.version`.
```

Comment:
Is there any documentation we can link the user to that enumerates the options they can set?

Reply:
That's a great question. I believe these will be documented per-product on their respective webhook pages - based on https://developer.hashicorp.com/hcp/docs/packer/webhooks

Linking to the HCP Packer page to start would be helpful but I don't know how fast it will become outdated once other products start to offer webhooks. 

Wondering if there should be a central webhooks page for HCP that gets updated as products enable webhooks, which can be linked to later on. What do you think @dadgar?

Reply:
Wilken is right, these will be documented per-product and I think the best solution would be to have a single page where we would link to those documentation. I think that makes total sense. 

I can work on adding a section to https://developer.hashicorp.com/hcp/docs/packer/webhooks, but before, I would like to confirm if that would address the concerns of this thread.

Reply:
What about having a generic web hooks documentation page that describes the capabilities and have links to each product page that integrates with web hooks for more details on events. That way it is scalable.

Something like: 
```
# Webhooks
How they work

## Setting them up examples
...

## Webhook Event Documentation

Packer Webhooks -> Link to existing page
HVS Webhooks -> Link to HVS
...
```

Reply:
This is a good option. Especially, if teams already update provider docs as they add platform support or other updates to resources this option would be ideal. 

 I considered it previously for a moment but thought that given the distance between the provider and the product docs themselves that teams not needing to change the provider could forget or not know now to update the docs. In my head the closer it is the the platform docs the more visibility it would get. 

Thinking out loud, users will probably find their way to the docs for their product of choice after clinking into anyone of the web events pages and searching. So maybe the docs becoming stale is not a big issue. A generic web hooks documentation page that describes the capabilities and links to product pages seems like a good tradeoff at the moment.

Reply:
Would this generic page be a guide?

Reply:
I think it would make more sense to add here so it can be reference for TF, CLI, API. 

https://github.com/hashicorp/hcp-docs

Reply:
Got it. I will add a new section to https://developer.hashicorp.com/hcp/docs/hcp/admin/projects/webhooks 
I think it also makes sense to add a section about the terraform resource.